### PR TITLE
LF: Fix encoder for Exception

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
Add support for `(from/to)AnyException`

part of #8020

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala [hunk 1]
```diff
@@ -637,6 +637,14 @@ private[daml] class EncodeV1(minor: LV.Minor) {
               .setExceptionType(excTy)
               .setExceptionExpr(exc)
           )
+        case EToAnyException(ty, body) =>
```
Comment (cocreature) [inline daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala:640]: Ugh, I hate the pattern match thingy.

**Review by cocreature — APPROVED** [review]
Thanks!

File: daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala [hunk 2]
```diff
@@ -637,6 +637,14 @@ private[daml] class EncodeV1(minor: LV.Minor) {
...
+          assertSince(LV.Features.exceptions, "Expr.FromAnyException")
```
Comment (sofiafaro-da) [inline daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala:641]:
```suggestion
          assertSince(LV.Features.exceptions, "Expr.ToAnyException")
```

Comment (remyhaemmerle-da) [reply daml-lf/encoder/src/main/scala/com/digitalasset/daml/lf/archive/testing/EncodeV1.scala]: Good catch!

**Pull request merged**